### ISSUE TITLE
Troubles with big images

### ISSUE THREAD

**Issue by hotgen** [prc5/react-zoom-pan-pinch#76]
So if I load a big image (like 2000x2000) into `TransformWrapper `it doesn't work as expected. Slow zoom, incorrect pan. I tried use `minScale `or put it into `overflow: auto` and it doesn't help. Any ideas?

**Comment by prc5** [comment]
> So if I load a big image (like 2000x2000) into `TransformWrapper `it doesn't work as expected. Slow zoom, incorrect pan. I tried use `minScale `or put it into `overflow: auto` and it doesn't help. Any ideas?

I will check this, thanks for reporting!

**Comment by zpetukhov** [comment]
@hotgen @prc5
Trying the component.
What I have found:
The main reason of all the numerous issues is the setState call when the animation is in progress.
Rough way to get rid of the issues: do not use onZoomChange, onPanning events.
@hotgen - try that, then see if many of the issues is solved.

Also. To get rid of scrollbars (for large images), I have to use this:
```
#viewerMain > div {
  max-width:100%;
  max-height:100%;
}
```
where viewerMain is the parent container of TransformWrapper.

Also. For large images, I found that it's makes sense to use settings like this:
```
       wheel={{step:100}}
       doubleClick={{step:10, animationTime: 0}}
```

PS. one of the animations is turned off by the following parameter:
pan={{lockAxisX, lockAxisY, panReturnAnimationTime: 0}}
Didn't found that in readme.

**Comment by Kipriz** [comment]
I faced the same issue with the slow zoom for big images (full screen). I think that the reason is in this calculation: 
https://github.com/prc5/react-zoom-pan-pinch/blob/f78c546414b13c61a68368db74ed9fd9fb2e185c/src/store/zoom/index.ts#L35-L37
As soon as wrapper's width is close to window's width, the scale factor becomes `0.99` that leads to very delta for the target scale.

Can you introduce an option to provide a scaleFactor function or always fix it to 0.2?
Wheel's step option works for big images, but it is necessary to calculate it depending on image size to have similar behavior.

**Comment by hamzamubeen-rebel** [comment]
I am Also facing this issue images with large resolution have glitchy animation when zooming in for the first time and then works fine after zooming in and zooming  out.. any workaround for this ?
